rmw_implementation/src/functions.cpp

```diff
@@ -61,17 +72,9 @@ get_library()
         ("failed to find shared library of rmw implementation. Searched " + env_var).c_str());
       return nullptr;
     }
-    try {
-      lib = new Poco::SharedLibrary(library_path);
-    } catch (Poco::LibraryLoadException & e) {
-      RMW_SET_ERROR_MSG(
-        ("failed to load shared library of rmw implementation. Exception: " +
-        e.displayText()).c_str());
-      return nullptr;
-    } catch (...) {
-      RMW_SET_ERROR_MSG("failed to load shared library of rmw implementation");
-      RMW_SET_ERROR_MSG(
-        ("failed to load shared library of rmw implementation: " + library_path).c_str());
+    rcutils_ret_t ret = rcutils_load_shared_library(lib, library_path.c_str());
+    if (ret != RCUTILS_RET_OK) {
+      RMW_SET_ERROR_MSG(("Cannot open library " + library_path).c_str());
       return nullptr;
```

Comment:
Like if you return here, isn't `lib` going to be leaked?

Reply:
used `delete` when a failure happened